### ISSUE TITLE
Improve the detection of user region to enable the correct regional list

### ISSUE THREAD

**Issue by MasterKia** [uBlockOrigin/uBlock-issues#2180]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

I've briefly discussed this with @Yuki2718 and @krystian3w, and it turns out that after the first installtion uBO checks for the browser language and selects the corresponding regional list for that language.

This approach is ineffective in my opinion; I can't speak for other languages (maybe maintainers could chime in?), but to this day I have yet to see someone changing their OS/browser language from English to Persian.

One reason for that is: Everything becomes _mirrored/reversed_ when you change to an RTL language (Persian, Arabic); so users get confused and stick to English.
See: https://user-images.githubusercontent.com/17685483/179814406-be4934d9-8c49-4deb-b4d8-cf5a10b78de1.png

It takes 5 clicks to find and enable a regional list after first install, so I'd say half of the users wouldn't bother doing that; If uBO's goal is to protect the users, then a better approach should be taken (because not all ads/trackers are covered by EL/EP).

IMO AdGuard's solution is effective but I presume it can have negative impact on performance(?).
> Language-specific filters will turn on depending on the site language.

For uBO; Instead of checking the browser language, I propose checking for the _user timezone_ and enable the regional list matching that timezone.
1) Check user timezone by the extension: maybe this needs extra permission or isn't technically possible.
2) [Check user timezone using javascript](https://stackoverflow.com/a/34602679): `Intl.DateTimeFormat().resolvedOptions().timeZone` returns the user timezone (tested on PC and mobile).

Example: https://masterkia.github.io

So after first install, uBO could make a request to `gorhill.github.io` and fetch the user timezone then enable the correct regional list.
And for privacy concerns (regarding sending request to GitHub); Well, filterlists are already hosted on GitHub. So GitHub knows which IP downloads them anyway.

### A specific URL where the issue occurs

https://masterkia.github.io

### Steps to Reproduce

.

### Expected behavior

.

### Actual behavior

.

### uBlock Origin version

1.43.0

### Browser name and version

Chrome

### Operating System and version

Windows

**Comment by gorhill** [comment]
> users get confused and stick to English

If their browser is configured to Farsi, why would they be confused by uBO's user interface being in Farsi?

**Comment by MasterKia** [comment]
I meant users might get confused when they change from English to Persian because suddenly everything is mirrored and reversed.

**Comment by MasterKia** [comment]
Also maintainers from other countries could say whether people configure their browser language or mostly stick to English.

**Comment by krystian3w** [comment]
I see bug if many countries have same timezone or very close (possible enable many EU lists worse than in uBo after detect language with similar method like Google Translator - so read filters List can enable italy supplement - e.g. I don't need Greek list updated Year Ago if no surf in these space of Internet).

**Comment by MasterKia** [comment]
> I see bug if many countries have same timezone

What do you see when visiting https://masterkia.github.io?

**Comment by MasterKia** [comment]
> I meant implementation based on _**only UTC- / UTC+**_

https://stackoverflow.com/a/34602679:
_Using an offset to calculate Timezone is a wrong approach, and you will always encounter problems. Time zones and daylight saving rules may change on several occasions during a year, and It's difficult to keep up with changes._

**Comment by MasterKia** [comment]
> unconvincing benefits.

The convincing _harm_ is that unless EasyList/EasyPrivacy covers all trackers/ads for every language, then non-English users won't be protected from trackers/ads; uBO'S goal is to _protect_ the users.

Non-English users would mostly visit non-English sites and EL/EP doesn't cover non-English, so those users could _convincingly_ benefit from the protection of regional lists.

**Comment by gorhill** [comment]
> Non-English users would mostly visit non-English sites and EL/EP doesn't cover non-English

Yes, hence why uBO enables automatically the lists matching the default language set in the browser.

**Comment by MasterKia** [comment]
And may I ask how do you know for sure that even half of the users configure the browser to their language?

**Comment by krystian3w** [comment]
"Time Zone" as `DateTimeFormat` method is not problem if you enforced Helsinki in Clock system settings and these no have other capitals like Tallinna, Vilna, Riika, Kiova, Tiraspol, Chișinău, Bukarest, Sofia, Ateena or Kairo.

If I correct checked, no possible spoof by these addons:

https://add0n.com/spoof-timezone.html 

https://chrome.google.com/webstore/detail/spoof-timezone/kcabmhnajflfolhelachlflngdbfhboe
https://addons.mozilla.org/addon/spoof-timezone/

**Comment by MasterKia** [comment]
> Unnecessary issue reports might arise because regional lists aren't enabled: https://github.com/uBlockOrigin/uAssets/issues/14368

I also want to add that EasyList's generic filters break some Persian websites, but that's understandable given the nature of `generic`.
What's not understandable is that they refuse to fix these breakages by closing issues and marking them as "spam".
https://github.com/easylist/easylist/issues/12152

Therefore I fix these breakages in my list.

So users that don't have the corresponding regional list enabled will suffer from these breakages and even if they report to uAssets, it'd be an extra burden for the volunteers because the issue has already been fixed in the regional list anyway.

**Comment by MasterKia** [comment]
> Using a time zone might be too inaccurate depending on which part of the world you live. E.g I, who live in Finland, would get a good amount of filter lists that are unnecessary to me: https://upload.wikimedia.org/wikipedia/commons/8/88/World_Time_Zones_Map.png (**many countries exist in the same time zone**).

The "tz database time zones" doesn't have this problem:
https://upload.wikimedia.org/wikipedia/commons/c/cb/2017a.png

**Comment by Yuki2718** [comment]
IDK how good the home page method works, but
> Another option could be that uBO asks after installation process, which languages / regions should be activated.

sounds the best - in the end there are user who visit more than one specific language sites and those who understand they need to enable according filter lists are ... rare.

**Comment by MasterKia** [comment]
If uBO asks in English then it's pointless for half of users who don't understand English well and they'll just hit ok to get rid of the annoying popup.

**Comment by MasterKia** [comment]
If uBO asks in English then it's pointless for users who don't understand English well, and they'll just hit ok to get rid of the annoying popup.

**Comment by Yuki2718** [comment]
Unless you install on trip lol.

**Comment by MasterKia** [comment]
@krystian3w I don't know much about extension development but can uBO execute this code on its own after installation?
`Intl.DateTimeFormat().resolvedOptions().timeZone`

**Comment by krystian3w** [comment]
IDK.

**Comment by MasterKia** [comment]
I asked an JS expert and he said: "everything you can execute in your console can be executed by a script or extension".

So PoC in uBO would be:
```
language = Intl.DateTimeFormat().resolvedOptions().timeZone;
return selectList(language);
````

And each list should have the correct tz timezone in https://github.com/gorhill/uBlock/blob/master/assets/assets.json

**Comment by Popedigoz** [comment]
Hello. Just wanted to say that I personally, have always manually opted to US-EN installation of any browser I've ever installed, even though I live in Europe and English is not my mother language. I always have everything on English, OS too.
